test/fields-relationship/e2e.spec.ts

```diff
@@ -322,6 +323,24 @@ describe('fields - relationship', () => {
     await expect(options).not.toContainText('exclude')
   })
 
+  test('should not query for a relationship when filterOptions returns false', async () => {
+    await payload.create({
+      collection: relationFalseFilterOptionSlug,
+      data: {
+        name: 'whatever',
+      },
+    })
+
+    await page.goto(url.create)
+
+    // select relationshipMany field that relies on siblingData field above
+    await page.locator('#field-relationshipManyFiltered .rs__control').click()
+
+    const options = page.locator('#field-relationshipManyFiltered .rs__menu')
+    await wait(10)
```

Comment:
Maybe check if the length of the entries in the rs__menu is > than what it should be to ensure everything is loaded. This should be less flaky in case wait(10) is not enough to guarantee it has loaded